templates/taxi/car_list.html

```diff
@@ -1,22 +1,26 @@
-{% extends "base.html" %}
-
-{% block content %}
-  <h1>
-    Car list
-    <a href="{% url 'taxi:car-create' %}" class="btn btn-primary link-to-page">
-      Create
-    </a>
-  </h1>
-  {% if car_list %}
-    <ul>
-      {% for car in car_list %}
-        <li>
-          <a href="{% url "taxi:car-detail" pk=car.id %} ">{{ car.id }}</a>
-          {{ car.model }} ({{ car.manufacturer.name }})
-        </li>
-      {% endfor %}
-    </ul>
-  {% else %}
-    <p>There are no cars in taxi</p>
-  {% endif %}
-{% endblock %}
+{% extends "base.html" %}
+
+{% block content %}
+  <h1>
+    Car list
+    <a href="{% url 'taxi:car-create' %}" class="btn btn-primary link-to-page">
+      Create
+    </a>
+  </h1>
+  {% if car_list %}
+    <form method="get", action="{% url 'taxi:search-cars' %}">
+      <input type="text", name="q", placeholder="Search cars (model)">
+      <button type="submit">Search</button>
+    </form>
+    <ul>
+      {% for car in car_list %}
+        <li>
+          <a href="{% url "taxi:car-detail" pk=car.id %} ">{{ car.id }}</a>

```

Comment:
There is an extra space at the end of the URL tag, which might cause issues with the link rendering. Consider removing it.